src/kernel/time/timer.c

```diff
@@ -31,6 +33,7 @@ void clock_tick_handler(int irq_num, void *dev_id) {
 	if (++cs->jiffies_cnt == (1 << SLOWDOWN_SHIFT)) {
 		cs->jiffies_cnt = 0;
 		cs->jiffies++;
+		timer_sched_cnt++;
```

Comment:
Probably, it would be better to do `timer_sched_cnt++` right before `lthread_launch(&clock_handler_lt);` since we want to handle only these clock ticks (i.e. from cs_jiffies)?

Reply:
By the way, these changes mean lthreads are working slower than interrupts somehow and do not count all clock interrupts properly, aren't them?

Reply:
> Probably, it would be better to do timer_sched_cnt++ right before lthread_launch(&clock_handler_lt) since we want to handle only these clock ticks (i.e. from cs_jiffies)?

Thanks, correct, that should be there.

> By the way, these changes mean lthreads are working slower than interrupts somehow and do not count all clock interrupts properly, aren't them?

Not exactly. Xen timer generates interrupts based on virtual time, i.e. CPU time consumed by particular OS. This requires us to catch up for passed real-time. More, tick is hardcoded for 10ms. So in Xen driver we'll call clock_tick_handler as many times as required to align with real time.